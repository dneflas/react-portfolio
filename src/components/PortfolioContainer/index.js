import React from "react";
import About from "../About";
import Portfolio from "../Portfolio";
import ContactForm from "../ContactForm";

function PortfolioContainer({ currentSection }) {
  // logic to conditionally render each section (About, Projects, Contact, and Resume)
  const renderSection = () => {
    if (currentSection === "About") {
      return <About />;
    } else if (currentSection === "Portfolio") {
      return <Portfolio />;
    } else if (currentSection === "Contact") {
      return <ContactForm />;
    }
  };

  return <main className="container">{renderSection()}</main>;
}

export default PortfolioContainer;
