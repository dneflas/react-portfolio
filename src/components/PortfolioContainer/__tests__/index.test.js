import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import PortfolioContainer from "..";

afterEach(cleanup);

describe("Portfolio container renders", () => {
  it("renders", () => {
    render(<PortfolioContainer />);
  });
  it("matches snapshot", () => {
    const { asFragment } = render(<PortfolioContainer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
