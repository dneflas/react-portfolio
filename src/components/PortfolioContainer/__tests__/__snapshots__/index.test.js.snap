// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio container renders matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="container"
  />
</DocumentFragment>
`;
